### ISSUE TITLE
Add a `DateAddYears` operation

### ISSUE THREAD

**Issue by evansd** [opensafely-core/databuilder#627]
We can generally approximate year arithmetic using day arithmetic with `int(years * 365.25)` days. In many cases this will be perfectly sufficient. (For instance, we only have dates of birth rounded to the start of the month, so trying to do anything more precise with these dates is pointless.) But in some cases we may need to calculate precise calendar year dates e.g. if a patient becomes eligible for X exactly one year after Y happens.

A slight wrinkle here is that, by default, different databases have different treatment for leap years e.g.
```
sqlite3
sqlite> SELECT date("2020-02-29", "1 year");
2021-03-01
```
```
just connect-to-mssql
1> SELECT DATEADD(year, 1, '2020-02-29');
2> GO
2021-02-28 00:00:00.000
```

The behaviour we want here is that given by SQLite i.e. `2020-02-29 + 1 year = 2021-03-01`

This gives us consistency with our `DateDifferenceInYears` operation, so I don't think any other option is viable.

For the ehrQL side of things we should probably hold our noses and create a pair of `add_years/subtract_years` methods.

These can then hopefully be replaced by the ["date delta"](https://github.com/opensafely-core/databuilder/issues/519) implementation.

**Comment by benbc** [comment]
@evansd I think this is obsolete, isn't it?

**Comment by evansd** [comment]
Closed by #828
